aixplain/modules/pipeline/designer/pipeline.py

```diff
@@ -0,0 +1,328 @@
+from typing import List, Type, Tuple, TypeVar
+
+from .base import Serializable, Node, Link
+from .nodes import (
+    Asset,
+    Decision,
+    Script,
+    Input,
+    Output,
+    Router,
+    Route,
+    BareReconstructor,
+    BareSegmentor,
+)
+from .enums import NodeType, RouteType, Operation, DataType
+
+
+T = TypeVar("T", bound="Asset")
+
+
+class Pipeline(Serializable):
```

Comment:
Should we give another class name to avoid confusion? Like `DesignerPipeline` or `PipelineArchitecture`?

Reply:
renamed to `DesignerPipeline`